actions/class.ItemContent.php

```diff
@@ -73,13 +79,20 @@ public function files()
                 $filters = array_map('trim', explode(',', $filterParameter));
             }
         }
-        $depth = $this->hasRequestParameter('depth') ? $this->getRequestParameter('depth') : 1;
+        $depth = $params['depth'] ?? 1;
+        $childrenLimit = $this->getPaginationLimit();
+        $childrenOffset = $params['childrenOffset'] ?? self::DEFAULT_PAGINATION_OFFSET;
+
         $resolver = new ItemMediaResolver($item, $itemLang);
-        $asset = $resolver->resolve($this->getRequestParameter('path'));
-        $data = $asset->getMediaSource()->getDirectory($asset->getMediaIdentifier(), $filters, $depth);
+        $asset = $resolver->resolve($params['path']);
+
+        $data = $asset->getMediaSource()->getDirectories(
+            new QueryObject($asset->getMediaIdentifier(), $filters, $depth, $childrenLimit, (int)$childrenOffset)
```

Comment:
I mentioned in the other PR to rename this object `QueryObject` to `DirectorySearchQuery` and not put it on tao-media-manager extension. 

Now we are coupling media manager with this extension. Wdyt?

Reply:
Did I miss something, but class is already located at tao-core, so no extra coupling there. 
BTW good idea about naming.

Reply:
Yes, the method is in tao-core already. You are right :)